zebra-state/src/service/check.rs

```diff
@@ -14,13 +14,13 @@ use zebra_chain::{
 use crate::{
     service::{
         block_iter::any_ancestor_blocks, check::difficulty::POW_ADJUSTMENT_BLOCK_SPAN,
-        finalized_state::FinalizedState, non_finalized_state::NonFinalizedState,
+        non_finalized_state::NonFinalizedState,
     },
     BoxError, PreparedBlock, ValidateContextError,
 };
 
 // use self as check
-use super::check;
+use super::{check, finalized_state::ZebraDb};
```

Comment:
Nitpick: I realise this is probably an auto-import thing, but this line is a special import that allows us to refer to functions in this module using `check::fn()`. They are more readable that way.

So I'm not sure we want any other imports in this list.